babeld/babel_main.c

```diff
@@ -315,9 +314,12 @@ babel_load_state_file(void)
                 zlog_err("Couldn't parse babel-state.");
             }
         }
-        close(fd);
-        fd = -1;
+        goto fini;
     }
+fini:
```

Comment:
can't we accomplish this without using a goto? change the second `if` to `else if`, drop the `fini` label and the `goto`s, and kill the `return` statement since it's implicit

Reply:
Yes we can, we I'd like to leave it as is since it makes the finalization of the function easier to read. For every finalization, let's push it at the end of the functions.

I'll resubmit an update without this comment Quentin, I'll include your other ones.

Reply:
I would agree if this was a large function or the code path involved was more complicated, but it's not. It's a simple if-else. In any case I don't feel strongly enough about it to block the fixes.

Reply:
thank you.